devices/Sim7080/Sim7080G.cs

```diff
@@ -0,0 +1,301 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+
+using System;
+using System.Diagnostics;
+using System.IO.Ports;
+using System.Text;
+using System.Threading;
+
+namespace IoT.Device.Sim7080
+{
+    /// <summary>
+    /// Sim7080G Multi-Band CAT-M(eMTC) and NB-IoT module
+    /// </summary>
+    public class Sim7080G
+    {
+        /// <summary>
+        /// The UART <see cref="SerialPort"/> for communication with the modem
+        /// </summary>
+        private readonly SerialPort _serialPort;
+
+        /// <summary>
+        /// Constructor
+        /// </summary>
+        /// <param name="serialPort">The UART <see cref="SerialPort"/> for communication with the modem</param>
+        public Sim7080G(SerialPort serialPort) => _serialPort = serialPort;
+
+        /// <summary>
+        /// Number of retries on failure
+        /// </summary>
+        public int Retry { get; set; } = 3;
+
+        /// <summary>
+        /// The network system mode for wireless and cellular network communication service
+        /// </summary>
+        public SystemMode SystemMode { get; private set; }
+
+        /// <summary>
+        /// Cellular network connection status
+        /// </summary>
+        public ConnectionStatus NetworkConnected { get; set; } = ConnectionStatus.Disconnected;
+
+        /// <summary>
+        /// MQTT endpoint connection status
+        /// </summary>
+        public ConnectionStatus EndpointConnected { get; set; } = ConnectionStatus.Disconnected;
+
+        /// <summary>
+        /// MQTT subscription topic connection status
+        /// </summary>
+        public ConnectionStatus TopicConnected { get; set; } = ConnectionStatus.Disconnected;
+
+        /// <summary>
+        /// Cellular Network Operator
+        /// </summary>
+        public string Operator { get; private set; } = "Unknown";
+
+        /// <summary>
+        /// The Public IP Address
+        /// </summary>
+        public string IPAddress { get; private set; } = "0.0.0.0";
```

Comment:
why not using an IPAddress class? Is there a good reason to have it as a string?

Reply:
Basically because the serial device returns a json acknowledgement which contains an array containing an IP string.

Reply:
fine for me. But you still should validate it's a valid IP at some point. What if you return something like "abdt.12" because it's not "0.0.0.0", your condition checking it's a valid IP will be correct. While actually you won't have a proper IP

Reply:
Make sense, will change that!

Reply:
I did some experiments.

When the SIM connects it gets an IP Address assign from the Network Operator. The SIM therefor has an IP Address.
When I call IPGlobalProperties.GetIPAddress() the IP is still 0.0.0.0, which tells me that the IP is not propagated to the device. 

All request to the internet are executed by the SIM and the response is reported back to the device. In other words, there is no direct connection/route to the internet. (https://m2msupport.net/m2msupport/http-testing-with-simcom-sim7070-sim7080-modules/).